### PR TITLE
Blog app - integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by udeaghad**
In this pull request, I have:'

- Made sure that the N+1 problem is solved when fetching all posts and their comments for a user
- Used Capybara to write integration tests for each view in your project.

Kindly review and approve

### CODE REVIEW

File: spec/features/user_show_spec.rb [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    #   ])
+
+    click_button('See all posts')
+    find(:button)
+  end
+
+  it 're-direct to post#show page when post title is clicked' do
+    #  mark_post =  Post.create(
+    # {user: @mark, title: 'Driving', text: 'text for 1', likes_counter: 2, comments_counter: 1}
+    # )
```
Comment (omar-muhamad) [inline spec/features/user_show_spec.rb]: - [ ] Please remove all commented code to make your code more readable and clean.

File: spec/features/post_index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  end
+
+  it 'displays how many likes a post has' do
+    expect(page).to have_content('Likes: 2')
+  end
+
+  it 're-directs user to post show page' do
+    click_link('Driving')
+    expect(page).to have_current_path(user_post_path(@mark, @mark_post))
+  end
```
Comment (omar-muhamad) [inline spec/features/post_index_spec.rb:49]: - [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=I%20can%20see%20a%20section%20for%20pagination%20if%20there%20are%20more%20posts%20than%20fit%20on%20the%20view.), Please add a test that asserts you can see a section for pagination if there are more posts than fit on the view.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @udeaghad and @MICHAELMUNAVU83,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- Gitflow is used correctly. ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by brytebee — APPROVED** [review]
Hi Team👋,

##

### Highlights💡
- Good commit message.✔️ 
- Good PR Summary.✔️
- Great schema design.✔️
- App functions as it should.✔️

##

# STATUS: APPROVED 🕺🥳 

Congratulations🍾🥳!! Your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 



##

###  **Optional Suggestions:**
- N/A

##

Good luck! 🎉 

------

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**